### PR TITLE
alias can access node_modules

### PR DESCRIPTION
**Pull request by dev-johnny-gh**

### CODE REVIEW

**Comment by dev-johnny-gh** [comment]
Is there anyone to review this pr?

**Comment by frostney** [comment]
Hi @duxiaofeng-github, thank you for your pull request. Could you please go more into what this changes are doing. From the first impression it feels to me a lot like it's trying to replicate behavior from `rollup-plugin-node-resolve`. Would you mind going into more details of where they differentiate?
Furthermore, since this would be a breaking change I believe this should live under a flag to be able to have the previous behavior for backwards compatibility.

**Comment by dev-johnny-gh** [comment]
Hi, @frostney. This pr let the plugin can make an alias to node_modules. for example:
I want to import a module from vue like: `import Vue from "vue";`,
but actually I want vue refer to `vue/dist/vue.esm.js` instead of `vue/dist/vue.js`.
In this case, the plugin can not process the node module's path correctly.
So I submit this pr to resolve the problem.

File: src/index.js [hunk 1]
```diff
@@ -88,6 +92,9 @@ export default function alias(options = {}) {
         return filePath + '.js';
       }
 
+      const nodeModuleFile = resolveNodeModules(updatedId);
+      if (nodeModuleFile) return nodeModuleFile;
```
Comment (dev-johnny-gh) [inline src/index.js:96]: As you can see. The plugin should resolve the file relative to project.
After that, if no file can be resolve, it try to resolve it with node_modules folder under `process.cwd()`.
So I think it should't cause compatibility issues.

Comment (dev-johnny-gh) [reply src/index.js:96]: So, you can write a alias like this:

> rollup({
>   entry: './src/index.js',
>   plugins: [alias({
>     vue: 'vue/dist/vue.esm.js'
>   })],
> });

to include the runtime template compiler in vue project.

Comment (xjchengo) [reply src/index.js:96]: I have some suggestions:
1. `process.cwd()` cannot use as project root. If people don't invoke `node` from project root, your code will wrong.
2. There may be other rollup plugins that can deal with module id. So resolving using node_modules folder as fallback can cause compatibility issue.

So you can try to add two config options: projectRoot and nodeModulesFallback
```
rollup({
  entry: './src/index.js',
  plugins: [alias({
    vue: 'vue/dist/vue.esm.js'
  }, projectRoot, nodeModulesFallback)],
});
```
projectRoot is used to indicate project root directory.
nodeModulesFallback default is false for backwards compatibility.

I think using `alias` in this way is also convenient:
```
rollup({
  entry: './src/index.js',
  plugins: [alias({
    vue: path.resolve('PROJECT_ROOT/node_modules', './vue/dist/vue.esm.js')
  })],
});
```

Comment (dev-johnny-gh) [reply src/index.js:96]: Sounds good. thx

Comment (dev-johnny-gh) [reply src/index.js:96]: I will close this pr. @xjchengo's comment is good enough to solve this issue.

**Pull request closed without merging**